src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -474,30 +476,33 @@ private File generateTargetFile(String originalFilename, String targetFolder, Bo
      * @param keepParentFolder    if true also creates the resources with the folder they are currently in (e.g. current/parent/* -> new/parent/*)
      * @throws IOException if the copying operation fails.
      */
-    public void copyResources(Resource[] resources, String prefix, String targetDirectoryPath, Boolean keepParentFolder) throws IOException {
+
+    public void copyResources(Resource[] resources, String prefix, Path targetDirectoryPath, Boolean keepParentFolder) throws IOException {
+
         for (Resource resource : resources) {
             // Replace windows separator with "/"
             String fileUrl = java.net.URLDecoder.decode(resource.getURL().toString(), StandardCharsets.UTF_8).replaceAll("\\\\", "/");
             // cut the prefix (e.g. 'exercise', 'solution', 'test') from the actual path
             int index = fileUrl.indexOf(prefix);
 
-            String targetFilePath = keepParentFolder ? fileUrl.substring(index + prefix.length()) : "/" + resource.getFilename();
+            // We use prefix.length() + 1 to also cut the "/" after the prefix
+            String targetFilePath = keepParentFolder ? fileUrl.substring(index + prefix.length() + 1) : resource.getFilename();
             targetFilePath = applySpecialFilenameReplacements(targetFilePath);
 
             if (isIgnoredDirectory(targetFilePath)) {
                 continue;
             }
-
-            Path copyPath = Path.of(targetDirectoryPath + targetFilePath);
-            File parentFolder = copyPath.toFile().getParentFile();
-            if (!parentFolder.exists()) {
-                Files.createDirectories(parentFolder.toPath());
+            // will append the targetFilePath to the targetDirectoryPath, targetFilePath cannot have a "/" prefix
+            Path copyPath = targetDirectoryPath.resolve(targetFilePath);
+            Path parentFolder = copyPath.getParent();
+            if (!Files.exists(parentFolder)) {
+                Files.createDirectories(parentFolder);
             }
 
             Files.copy(resource.getInputStream(), copyPath, REPLACE_EXISTING);
             // make gradlew executable
             if (targetFilePath.endsWith("gradlew")) {
-                copyPath.toFile().setExecutable(true);
+                Files.getPosixFilePermissions(copyPath).add(PosixFilePermission.OWNER_EXECUTE);
```

Comment:
Please [take a look](https://github.com/ls1intum/Artemis/pull/5992/files) at the Codacy warning here.